packages/table/src/TableCell.tsx

```diff
@@ -1,30 +1,38 @@
 import React from 'react';
 import classNames from 'classnames';
-import { useTableContext } from './TableContext';
 import { OnTableClickEvent } from './types/OnTableClickEvent';
 import { Cell, IdType } from './types/ReactTable';
 
 type Props<T extends IdType> = {
   id?: string;
   cell: Cell<T>;
+  scrollable?: boolean;
+  getCellProps: (cell: Cell<T>) => React.HTMLAttributes<HTMLTableCellElement>;
+
   children: React.ReactNode;
   onCellClick?: (event: OnTableClickEvent<HTMLElement, T>) => void;
 } & React.HTMLAttributes<HTMLElement>;
 
-const TableCell = <T extends IdType>({ cell, children, onCellClick, ...rest }: Props<T>): JSX.Element => {
-  const { scrollable } = useTableContext();
+const TableCell = <T extends IdType>({
+  cell,
+  scrollable,
+  children,
+  onCellClick,
+  getCellProps,
+  ...rest
+}: Props<T>): JSX.Element => {
   const { className, disableClick, stickyLeft, stickyRight } = cell.column;
 
   const isFixedWidth = scrollable && !className;
+  const cellClasses = classNames(className || '', {
```

Comment:
I think we can move this to line 50 where all of the classes are reconciled. That way this calculation will only have to happen when that function is called